dartagnan/src/main/antlr4/LitmusC.g4

```diff
@@ -239,6 +261,28 @@ c11Mo returns [String mo]
     |   MoSeqCst    {$mo = C11.MO_SC;}
     ;
 
+openCLSpace returns [String space]
+    :   OpenCLGlobalSpace  {$space = OpenCL.GLOBAL_SPACE;}
+    |   OpenCLLocalSpace   {$space = OpenCL.LOCAL_SPACE;}
+    ;
+
+openCLScope returns [String scope]
+    :   OpenCLMemoryScopeWI    {$scope = OpenCL.WORK_ITEM;}
+    |   OpenCLMemoryScopeWG    {$scope = OpenCL.WORK_GROUP;}
+    |   OpenCLMemoryScopeDEV   {$scope = OpenCL.DEVICE;}
+    |   OpenCLMemoryScopeALL   {$scope = OpenCL.ALL;}
```

Comment:
Is all intended for host/device share memory? We are not using this at all at the moment, right? (I am not suggesting getting rid of it if the model uses it)

Reply:
We are using them now, we have litmus from herd/overhauling testing about memory scope.

Reply:
What is the difference between `device` and `all`? I would expect the first to allow all threads in the same device to communicate, meaning that only `all` would mean we are talking about sync between threads in different devices, or host/device. I don't think our litmus format allows us to define host code. Can we specify that threads belong to different devices?

Reply:
Yes now the opencl litmus format only allow device code. We are using the `P0@wg 0, dev 0` to specify which device the thread belong to.